infoschema/tables.go

```diff
@@ -792,8 +792,12 @@ func dataForColumns(ctx sessionctx.Context, schemas []*model.DBInfo) [][]types.D
 func dataForColumnsInTable(schema *model.DBInfo, tbl *model.TableInfo) [][]types.Datum {
 	var rows [][]types.Datum
 	for i, col := range tbl.Columns {
+		var charMaxLen, charOctLen, numericPrecision, numericScale, datetimePrecision interface{}
```

Comment:
Why not use int instead of interface{}? Are there any benefits of using interface{} here?

Reply:
using `interface{}`, we can got `NULL` result~

Reply:
interface is really slow.

Reply:
yes~~we should void using `interface{}` as we can, but in this case is special, `types.MakeDatums` accept `args ...interface{}` as arguments.

cast to interface always happen, so direct using `interface{}` local variable is simple.

we also can use `*int` local varible instead of `interface{}`, but that will make us step into a well-known trap in go https://play.golang.org/p/qw6UJhvZcrR